ports/zephyr/main.c

```diff
@@ -174,17 +180,23 @@ int real_main(void) {
     machine_pin_deinit();
     #endif
 
+    #if MICROPY_PY_THREAD
+    mp_thread_deinit();
+    gc_sweep_all();
+    gc_collect();
```

Comment:
Why exactly do you need to call `gc_sweep_all()` and `gc_collect()` here?  You shouldn't really call `gc_collect()` after `gc_sweep_all()`, that's not really well defined I don't think...

Reply:
Somehow I thought that was required. I'll look into this and fix it.

Reply:
This bit still needs investigation.

Reply:
Done. I need to call `gc_collect()` after `mp_thread_deinit()` otherwise soft-reset starts giving issues. But now `gc_sweep_all()` is called after (not before) and added a call to `mp_deinit()` to be in-line with the rest of the ports.